app/src/main/java/com/duckduckgo/app/browser/downloader/NetworkFileDownloader.kt

```diff
@@ -34,8 +39,58 @@ class NetworkFileDownloader @Inject constructor(private val context: Context, pr
             return
         }
 
-        val guessedFileName = filenameExtractor.extract(pendingDownload)
+        when (val extractionResult = filenameExtractor.extract(pendingDownload)) {
+            is FilenameExtractor.FilenameExtractionResult.Success -> downloadFile(pendingDownload, extractionResult.filename, callback)
+            is FilenameExtractor.FilenameExtractionResult.Failure -> tryDownloadFallback(pendingDownload, callback)
+        }
+
+    }
+
+    private fun tryDownloadFallback(pendingDownload: PendingFileDownload, callback: FileDownloader.FileDownloadListener) {
+        val downloadCall = fileService.downloadFile(pendingDownload.url)
+        downloadCall?.enqueue(object : Callback<Void> {
+            override fun onResponse(call: Call<Void>, response: Response<Void>) {
+                if (response.isSuccessful) {
+                    var updatedPendingDownload = pendingDownload.copy()
+                    if (response.headers().get("content-type") != null) {
+                        val mimeType = response.headers().get("content-type")
+                        updatedPendingDownload = updatedPendingDownload.copy(mimeType = mimeType)
+                    }
+
+                    if (response.headers().get("content-disposition") != null) {
+                        val contentDisposition = response.headers().get("content-disposition")
+                        updatedPendingDownload = updatedPendingDownload.copy(contentDisposition = contentDisposition)
+                    }
```

Comment:
```suggestion
                    var updatedPendingDownload = pendingDownload.copy(
                      mimeType = response.headers().get("content-type"),
                      contentDisposition = response.headers().get("content-disposition")
                     )
```

No need to make so many copies, `mimeType` and `contentDisposition` are already nullable.